src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1454,11 +1456,9 @@ bool NetworkOPsImp::beginConsensus (uint256 const& networkClosed)
         return false;
     }
 
-    assert (prevLedger->getHash () == closingInfo.parentHash);
+    assert (prevLedger->info().hash == closingInfo.parentHash);
     assert (closingInfo.parentHash ==
-            m_ledgerMaster.getClosedLedger ()->getHash ());
-
-    prevLedger->setImmutable (app_.config());
```

Comment:
I dug into this a bit, and it does look OK. `setImmutable` is called through `setAccepted`, and `setAccepted` is called when we load a ledger or when consensus accepts a ledger.